django/forms/widgets.py

```diff
@@ -61,6 +63,64 @@ class MediaOrderConflictWarning(RuntimeWarning):
     pass
 
 
+@html_safe
+class MediaAsset:
+    element_template = "{path}"
+
+    def __init__(self, path, **attributes):
+        self._path = path
+        self.attributes = attributes
+
+    def __eq__(self, other):
+        return (self.__class__ is other.__class__ and self.path == other.path) or (
+            isinstance(other, str) and self._path == other
+        )
```

Comment:
It may make sense to allow embedding the same script with different attributes, so I'm not sure if comparing `path` only is sufficient.

I think this is better:
https://github.com/matthiask/django-js-asset/blob/cd6be3edafb7864cb3bddf1608c039be25040c10/js_asset/js.py#L85-L87

- If RHS doesn't have attributes only compare the path
- If RHS has attributes they have to match

Reply:
Thanks for mentioning this. I was contemplating this, but it's a bit tricky. I co-authored the algorithm that ensure an intact dependency order in Django's media assets. It highly relies on equality checks and will grind to a hold on larger projects when you start comparing larger dictionaries.

Even if there is a valid use case (of which I am not aware), subclassing would be the most viable solution. Out of the box, Django needs to maintain reasonably fast render times. Default behaviors should benefit the less experienced users, and pros like you and I can subclass to build their elaborate features ☺️ 

I will add a comment to the implementation, so can follow the reasoning and make adjustments if they desire too.

Reply:
[I know ☺️](https://github.com/django/django/pull/11019), that was an amazing experience and much work has happened before and after - I'm not claiming the credit here. I just thought it was interesting.

I'm not sure the dictionary comparison will be a problem in the general case. I think treating the attributes as a part of the equality would be less surprising, and people could subclass the object or provide their own when they have slowdowns in their projects. That's just my opinion obviously.

Reply:
Oh, the same 🙈
![Sad Shame GIF](https://github.com/user-attachments/assets/780ce2a3-1009-4fa8-ac30-c4de2325340c)

I am happy that you are here again, to clean up my mistakes if need be ❤️ 

That aside, I believe we can't have it both ways, since the `__hash__` implementation is only path-based too. 

Currently, I see a stronger argument for the performance case. However, I am currently also not aware of a use case for injecting a resource twice. Except for preloading, but you'd currently need to create a subclass where you'd set `rel=preload` anyway.

Maybe @claudep could share some light on his original intentions here?